merkletree.go

```diff
@@ -111,6 +125,7 @@ func (m *MerkleTree) insertNode(key []byte, toAdd *userLeafNode) {
 	depth := 0
```

Comment:
The argument name should be `index`, not `key`.

It might be nice to do `indexBits := util.ToBits(index)`, as above, rather than `GetNthBit`, for consistency.